### PR TITLE
Add doxygen documentation for CONNECTION STATUS CALLBACK STATUS an REASON

### PR DESCRIPTION
**Pull request by ewertons**
Also include documentation for missing connection status reasons IOTHUB_CLIENT_CONNECTION_NO_PING_RESPONSE and IOTHUB_CLIENT_CONNECTION_QUOTA_EXCEEDED.

 

# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

The enums IOTHUB_CLIENT_CONNECTION_STATUS and IOTHUB_CLIENT_CONNECTION_STATUS_REASON do not have (doxygen) documentation in code.

# Description of the solution
 
Added doxygen-style documentation for both enumerations.

### CODE REVIEW

**Review by danewalton — APPROVED** [review]

File: iothub_client/inc/iothub_client_core_common.h [hunk 1]
```diff
@@ -161,6 +166,54 @@ extern "C"
...
+    *                     to the Azure IoT Hub and it will no longer try.
+    *               HTTP: Not applicable.
+    *           @c IOTHUB_CLIENT_CONNECTION_NO_NETWORK
+    *               MQTT: An MQTT CONNECT packet failed to be sent to the Azure IoT Hub for any reason.
+    *               AMQP: A network connection issue was detected, which includes socket errors,
+    *                     failures on AMQP ATTACH to CBS link (for authentication).
+    *               HTTP: Not applicable.
+    *           @c IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR
+    *               MQTT: A telemetry message timed out receiving a PUBACK from the Azure IoT Hub,
+    *                     or there was an error sending a PUBACK or PUBREC to Azure IoT Hub or
```
Comment (vaavva) [inline iothub_client/inc/iothub_client_core_common.h:198]:
```suggestion
    *                     there was an error sending a PUBACK or PUBREC to Azure IoT Hub, or
```

**Review by vaavva — APPROVED** [review]

**Review by JoseSanchez7 — APPROVED** [review]

File: iothub_client/inc/iothub_client_core_common.h [hunk 2]
```diff
@@ -161,6 +166,54 @@ extern "C"
     IOTHUB_CLIENT_CONNECTION_QUOTA_EXCEEDED                \
 
     /** @brief Enumeration passed to the application callback indicating reason that connection was unsuccessful.
+    *   @remark Each of the values of @c IOTHUB_CLIENT_CONNECTION_STATUS_REASON has a specific semanthics
```
[the remark this answers is not in the document]
Comment (ewertons) [reply iothub_client/inc/iothub_client_core_common.h:169]: What would I do without you both? @vaavva @danewalton

[the remark this answers is not in the document]
Comment (ewertons) [reply iothub_client/inc/iothub_client_core_common.h:169]: Fixed.

File: iothub_client/inc/iothub_client_core_common.h [hunk 3]
```diff
@@ -161,6 +166,54 @@ extern "C"
...
+    *           depending on the transport protocol used. Here is a brief description of he reasons for each
```
[the remark this answers is not in the document]
Comment (ewertons) [reply iothub_client/inc/iothub_client_core_common.h:170]: Fixed.

File: iothub_client/inc/iothub_client_core_common.h (hunk 1 above)
Comment (ewertons) [reply iothub_client/inc/iothub_client_core_common.h:198]: Fixed.

**Pull request merged**